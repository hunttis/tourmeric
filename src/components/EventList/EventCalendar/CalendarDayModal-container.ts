import { connect } from 'react-redux';
import { compose } from 'redux';
import { withRouter } from 'react-router-dom';
import _ from 'lodash';

import { setReturnLocation } from '../../../actions/eventEditorActions';
import { CalendarDayModal } from './CalendarDayModal';
import { ReduxState } from '../../../models/ReduxState';

export default compose(
  connect((state: ReduxState) => ({
    openinghoursexceptions: state.firebase.data.openinghoursexceptions,
    isAdmin: _.get(state, 'firebase.profile.role', 'user') === 'admin',
    setReturnLocation: (returnLocation: string) => setReturnLocation(returnLocation),
  })),
)(withRouter<any, any>(CalendarDayModal));
