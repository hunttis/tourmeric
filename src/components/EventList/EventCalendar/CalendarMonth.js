import React from 'react';
import moment from 'moment/min/moment-with-locales';
import momentEn from 'moment';
import _ from 'lodash';

export const CalendarMonth = ({ chunkedCalendar, categories, clickDay, openinghoursexceptions, settings }) => chunkedCalendar.map((week, weekIndex) => (
  <div key={`calendar-week-${weekIndex}`} className="column is-12 columns is-marginless">
    {week.map((day, dayIndex) => {

      let dayClass = '';
      const today = moment();
      const dayMoment = moment(day.dayLink, 'YYYY/MM/DD');
      if (dayMoment.isSame(today, 'day')) {
        dayClass = 'today-card';
      } else if (dayMoment.isBefore(today, 'day')) {
        dayClass = 'past-card';
      } else if (dayMoment.isAfter(today, 'day')) {
        dayClass = 'future-card';
      }

      if (day.empty) {
        return (
          <div key={`calendar-day-${dayIndex}`} className="column is-paddingless is-marginless is-hidden-mobile" />
        );
      }

      const exception = _.get(openinghoursexceptions, dayMoment.format('YYYY-MM-DD'));

      const englishDayName = momentEn(dayMoment.format('DD-MM-YYYY'), 'DD-MM-YYYY').format('dddd').toLowerCase();
      const hasNoOpeningHoursNormally = _.isEmpty(_.get(settings, `openingHours.${englishDayName}`));
      const closedThisDay = (exception && !exception.open) || (hasNoOpeningHoursNormally && (!exception || !exception.open));

      return (
        <div
          key={`calendar-day-${dayIndex}`}
          className={`column is-paddingless is-marginless ${_.isEmpty(day.eventsForDay) && 'is-hidden-mobile'}`}
        >

          <div className={`card calendar-day ${dayClass} ${closedThisDay && 'strikeover'}`} onClick={() => { clickDay(day); }}>
            <div className="card-header">
              <div className="card-header-title calendar-title">
                <div className="level calendar-cardtitle">
                  <div className="level-left">
                    <span className="is-hidden-mobile">
                      {day.day}
                    </span>
                    <span className="is-hidden-tablet">
                      {moment(day.dayLink, 'YYYY/MM/DD').format('dddd DD.MM.YYYY')}
                    </span>
                  </div>
                  <div className="level-right has-text-right has-text-info is-hidden-mobile">
                    {moment.weekdaysShort()[day.dayOfWeek]}
                  </div>
                </div>
              </div>
            </div>
            <div className="card-content is-paddingless">
              <div className="is-inline-flex">
                <div className="calendar-card-spacer calendar-image" />
                {!closedThisDay && day.ongoingEventsForDay.map((eventEntry, index) => {
                  const event = eventEntry.value;
                  const eventLogo = categories[event.category].image;
                  return (
                    <img key={`event-img-${index}`} className="image is-24x24 is-rounded calendar-image ongoing-event-image" src={eventLogo} alt="" />
                  );
                })}
                {day.eventsForDay.map((eventEntry, index) => {
                  const event = eventEntry.value;
                  const eventLogo = categories[event.category].image;
                  return (
                    <img key={`event-img-${index}`} className="image is-24x24 calendar-image" src={eventLogo} alt="" />
                  );
                })}
              </div>
            </div>
          </div>
        </div>
      );
    })}
  </div>
));
