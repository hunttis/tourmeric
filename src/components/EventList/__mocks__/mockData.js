export const mockEventId = 'mockEventId1';

export const mockUserId = 'mockUserId';

export const mockProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{ displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890' }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
};

export const mockUnorderedEvents = {
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents = [
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png' },
  };

export const mockParticipations = { mockEventId1: { mockUserId: { comment: 'Jee',
  date: '2018-06-23T09:40:26.992Z',
  firstName: 'Test',
  lastName: 'Person',
  userId: 'mockUserId' } } };

export const mockCategories = { category1: { abbreviation: 'WH40K',
  logo: '1527227370219137123',
  name: 'Warhammer 40,000',
  type: 'Miniatures' },
category2: { abbreviation: 'MtG',
  logo: '152722965771313789',
  name: 'Magic: The Gathering',
  type: 'TCG' },
'-LAlvCmIWNzYKpa7-7HI': { abbreviation: 'Lautapelit',
  logo: '152719272536668394',
  name: 'Lautapelit / Board games' },
'-LCmESny8m9JVEFQ_qOt': { abbreviation: 'PKM',
  logo: '152722747904159438',
  name: 'Pok√©mon',
  type: 'TCG' },
'-LDImTAGxkzrorL0tQdi': { abbreviation: 'VG',
  logo: '152719372155055672',
  name: 'Videogames' },
'-LDIw-1LFgiM0EJZGfus': { abbreviation: 'Necromunda',
  logo: '15271953483568546',
  name: 'Necromunda',
  type: 'Miniatures' } };
