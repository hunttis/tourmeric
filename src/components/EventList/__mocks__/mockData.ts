import { FirebaseProfile, Participation, User, ProviderData, Location } from '~/models/ReduxState';
import { TourmericEvent } from '~/models/Events';
import { Settings } from '~/models/Settings';
import { UploadedFile, Category } from '~/models/Category';

export const mockEventId = 'mockEvent1';

export const mockUserId = 'mockUserId';

export const mockLocation: Location = {
  pathname: '/events/2018/10',
  search: 'search',
  hash: 'hash',
  key: 'key',
};

export const mockProviderData: ProviderData = {
  displayName: 'MockName',
  email: 'mock@userton.com',
  photoURL: 'http://url',
  providerId: 'facebook',
  uid: '1234567890',
};

export const mockUser: User = {
  avatarUrl: 'http://www.google.fi/google.png',
  displayName: 'Mock Userton',
  firstName: 'Mock',
  lastName: 'Userton',
  providerData: mockProviderData,
  email: 'mock@userton.com',
  username: 'Mockie',
  dciNumber: '123456',
  role: 'user',
  active: true,
};

export const mockProfile: FirebaseProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{
    displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890',
  }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent: TourmericEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
  eventType: 'single',
};

export const mockUnorderedEvents: {[key: string]: TourmericEvent} = {
  mockEvent1: mockSingleEvent,
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents: {key: string, value: TourmericEvent}[] = [
  { key: 'mockEvent1',
    value: mockSingleEvent },
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings: Settings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos: {[key: string]: UploadedFile} =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png', name: 'image' },
  };

export const mockParticipations: {[key: string]: Participation} = {
  mockEventId1: {
    mockUserId: {
      comment: 'Jee',
      date: '2018-06-23T09:40:26.992Z',
      firstName: 'Test',
      lastName: 'Person',
      userId: 'mockUserId',
    },
  },
};

export const mockCategories: {[key: string]: Category} = {
  category1: {
    abbreviation: 'WH40K',
    logo: '1527227370219137123',
    name: 'Warhammer 40,000',
    type: 'Miniatures',
  },
  category2: {
    abbreviation: 'MtG',
    logo: '152722965771313789',
    name: 'Magic: The Gathering',
    type: 'TCG',
  },
  '-LAlvCmIWNzYKpa7-7HI': {
    abbreviation: 'Lautapelit',
    logo: '152719272536668394',
    name: 'Lautapelit / Board games',
    type: 'Boardgame',
  },
  '-LCmESny8m9JVEFQ_qOt': {
    abbreviation: 'PKM',
    logo: '152722747904159438',
    name: 'Pok√©mon',
    type: 'TCG',
  },
  '-LDImTAGxkzrorL0tQdi': {
    abbreviation: 'VG',
    logo: '152719372155055672',
    name: 'Videogames',
    type: 'Videogames',
  },
  '-LDIw-1LFgiM0EJZGfus': {
    abbreviation: 'Necromunda',
    logo: '15271953483568546',
    name: 'Necromunda',
    type: 'Miniatures',
  },
};
