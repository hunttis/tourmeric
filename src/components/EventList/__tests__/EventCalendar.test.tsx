jest.mock('history')
jest.mock('../../Common/DocumentUtils');

import React from 'react';
import { shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import EventCalendar from '../EventCalendar/EventCalendar';
import { mockEvents, mockCategories, mockSettings } from '../__mocks__/mockData';


describe('EventList tests', () => {

  it('Compares EventList to snapshot', () => {

    const history = createBrowserHistory();

    const eventList = shallow(<EventCalendar
      settings={mockSettings}
      events={mockEvents}
      eventsongoing={mockEvents}
      categories={mockCategories}
      activeLanguage='fi'
      location={{ pathname: '/events/2018/10' }}
      history={history}
      openinghoursexceptions={{}}
      isAdmin={false}
      setReturnLocation={() => {}}
    />);

    expect(eventList).toMatchSnapshot();
  });
});
