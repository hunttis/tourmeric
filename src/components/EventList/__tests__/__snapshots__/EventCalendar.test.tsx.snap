// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList tests Compares EventList to snapshot 1`] = `
<section
  className="section"
>
  <div
    className="container"
  >
    <h1
      className="title"
    >
      <Translate
        id="nextevents"
      />
    </h1>
    <div
      className="columns is-multiline"
    >
      <div
        className="column is-12"
      >
        <Translate
          id="filtereventsbychoosingcategories"
        />
        <Translate>
          <Component />
        </Translate>
      </div>
      <div
        className="column is-12 is-mobile has-text-centered"
      >
        <div
          className="is-one-quarter-desktop is-one-quarter-tablet is-one-third-mobile has-text-centered is-inline-flex"
          key="categoryfilter-category1"
        >
          <button
            className="button is-grouped  is-hidden-mobile image-square"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-48x48"
            />
          </button>
          <button
            className="button is-grouped  is-hidden-tablet"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-24x24"
            />
          </button>
        </div>
        <div
          className="is-one-quarter-desktop is-one-quarter-tablet is-one-third-mobile has-text-centered is-inline-flex"
          key="categoryfilter-category2"
        >
          <button
            className="button is-grouped  is-hidden-mobile image-square"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-48x48"
            />
          </button>
          <button
            className="button is-grouped  is-hidden-tablet"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-24x24"
            />
          </button>
        </div>
        <div
          className="is-one-quarter-desktop is-one-quarter-tablet is-one-third-mobile has-text-centered is-inline-flex"
          key="categoryfilter--LAlvCmIWNzYKpa7-7HI"
        >
          <button
            className="button is-grouped  is-hidden-mobile image-square"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-48x48"
            />
          </button>
          <button
            className="button is-grouped  is-hidden-tablet"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-24x24"
            />
          </button>
        </div>
        <div
          className="is-one-quarter-desktop is-one-quarter-tablet is-one-third-mobile has-text-centered is-inline-flex"
          key="categoryfilter--LCmESny8m9JVEFQ_qOt"
        >
          <button
            className="button is-grouped  is-hidden-mobile image-square"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-48x48"
            />
          </button>
          <button
            className="button is-grouped  is-hidden-tablet"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-24x24"
            />
          </button>
        </div>
        <div
          className="is-one-quarter-desktop is-one-quarter-tablet is-one-third-mobile has-text-centered is-inline-flex"
          key="categoryfilter--LDImTAGxkzrorL0tQdi"
        >
          <button
            className="button is-grouped  is-hidden-mobile image-square"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-48x48"
            />
          </button>
          <button
            className="button is-grouped  is-hidden-tablet"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-24x24"
            />
          </button>
        </div>
        <div
          className="is-one-quarter-desktop is-one-quarter-tablet is-one-third-mobile has-text-centered is-inline-flex"
          key="categoryfilter--LDIw-1LFgiM0EJZGfus"
        >
          <button
            className="button is-grouped  is-hidden-mobile image-square"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-48x48"
            />
          </button>
          <button
            className="button is-grouped  is-hidden-tablet"
            onClick={[Function]}
          >
            <img
              alt=""
              className="image is-24x24"
            />
          </button>
        </div>
      </div>
      <div
        className="column is-6"
      >
        <h1
          className="title"
        >
          Lokakuu
        </h1>
      </div>
      <div
        className="column is-6"
      >
        <div
          className="buttons has-addons is-right"
        >
          <button
            className="button"
            onClick={[Function]}
          >
            <Translate
              id="previousmonth"
            />
          </button>
          <button
            className="button"
            onClick={[Function]}
          >
            <Translate
              id="nextmonth"
            />
          </button>
        </div>
      </div>
      <Component
        categories={
          Object {
            "-LAlvCmIWNzYKpa7-7HI": Object {
              "abbreviation": "Lautapelit",
              "logo": "152719272536668394",
              "name": "Lautapelit / Board games",
              "type": "Boardgame",
            },
            "-LCmESny8m9JVEFQ_qOt": Object {
              "abbreviation": "PKM",
              "logo": "152722747904159438",
              "name": "PokÃ©mon",
              "type": "TCG",
            },
            "-LDImTAGxkzrorL0tQdi": Object {
              "abbreviation": "VG",
              "logo": "152719372155055672",
              "name": "Videogames",
              "type": "Videogames",
            },
            "-LDIw-1LFgiM0EJZGfus": Object {
              "abbreviation": "Necromunda",
              "logo": "15271953483568546",
              "name": "Necromunda",
              "type": "Miniatures",
            },
            "category1": Object {
              "abbreviation": "WH40K",
              "logo": "1527227370219137123",
              "name": "Warhammer 40,000",
              "type": "Miniatures",
            },
            "category2": Object {
              "abbreviation": "MtG",
              "logo": "152722965771313789",
              "name": "Magic: The Gathering",
              "type": "TCG",
            },
          }
        }
        chunkedCalendar={
          Array [
            Array [
              Object {
                "day": "01",
                "dayLink": "2018/10/01",
                "dayName": "Monday",
                "dayOfWeek": 1,
                "dayString": "01-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "02",
                "dayLink": "2018/10/02",
                "dayName": "Tuesday",
                "dayOfWeek": 2,
                "dayString": "02-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "03",
                "dayLink": "2018/10/03",
                "dayName": "Wednesday",
                "dayOfWeek": 3,
                "dayString": "03-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "04",
                "dayLink": "2018/10/04",
                "dayName": "Thursday",
                "dayOfWeek": 4,
                "dayString": "04-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "05",
                "dayLink": "2018/10/05",
                "dayName": "Friday",
                "dayOfWeek": 5,
                "dayString": "05-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "06",
                "dayLink": "2018/10/06",
                "dayName": "Saturday",
                "dayOfWeek": 6,
                "dayString": "06-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "07",
                "dayLink": "2018/10/07",
                "dayName": "Sunday",
                "dayOfWeek": 0,
                "dayString": "07-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
            ],
            Array [
              Object {
                "day": "08",
                "dayLink": "2018/10/08",
                "dayName": "Monday",
                "dayOfWeek": 1,
                "dayString": "08-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "09",
                "dayLink": "2018/10/09",
                "dayName": "Tuesday",
                "dayOfWeek": 2,
                "dayString": "09-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "10",
                "dayLink": "2018/10/10",
                "dayName": "Wednesday",
                "dayOfWeek": 3,
                "dayString": "10-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "11",
                "dayLink": "2018/10/11",
                "dayName": "Thursday",
                "dayOfWeek": 4,
                "dayString": "11-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "12",
                "dayLink": "2018/10/12",
                "dayName": "Friday",
                "dayOfWeek": 5,
                "dayString": "12-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "13",
                "dayLink": "2018/10/13",
                "dayName": "Saturday",
                "dayOfWeek": 6,
                "dayString": "13-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "14",
                "dayLink": "2018/10/14",
                "dayName": "Sunday",
                "dayOfWeek": 0,
                "dayString": "14-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
            ],
            Array [
              Object {
                "day": "15",
                "dayLink": "2018/10/15",
                "dayName": "Monday",
                "dayOfWeek": 1,
                "dayString": "15-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "16",
                "dayLink": "2018/10/16",
                "dayName": "Tuesday",
                "dayOfWeek": 2,
                "dayString": "16-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "17",
                "dayLink": "2018/10/17",
                "dayName": "Wednesday",
                "dayOfWeek": 3,
                "dayString": "17-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "18",
                "dayLink": "2018/10/18",
                "dayName": "Thursday",
                "dayOfWeek": 4,
                "dayString": "18-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "19",
                "dayLink": "2018/10/19",
                "dayName": "Friday",
                "dayOfWeek": 5,
                "dayString": "19-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "20",
                "dayLink": "2018/10/20",
                "dayName": "Saturday",
                "dayOfWeek": 6,
                "dayString": "20-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "21",
                "dayLink": "2018/10/21",
                "dayName": "Sunday",
                "dayOfWeek": 0,
                "dayString": "21-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
            ],
            Array [
              Object {
                "day": "22",
                "dayLink": "2018/10/22",
                "dayName": "Monday",
                "dayOfWeek": 1,
                "dayString": "22-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "23",
                "dayLink": "2018/10/23",
                "dayName": "Tuesday",
                "dayOfWeek": 2,
                "dayString": "23-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "24",
                "dayLink": "2018/10/24",
                "dayName": "Wednesday",
                "dayOfWeek": 3,
                "dayString": "24-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "25",
                "dayLink": "2018/10/25",
                "dayName": "Thursday",
                "dayOfWeek": 4,
                "dayString": "25-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "26",
                "dayLink": "2018/10/26",
                "dayName": "Friday",
                "dayOfWeek": 5,
                "dayString": "26-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "27",
                "dayLink": "2018/10/27",
                "dayName": "Saturday",
                "dayOfWeek": 6,
                "dayString": "27-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "28",
                "dayLink": "2018/10/28",
                "dayName": "Sunday",
                "dayOfWeek": 0,
                "dayString": "28-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
            ],
            Array [
              Object {
                "day": "29",
                "dayLink": "2018/10/29",
                "dayName": "Monday",
                "dayOfWeek": 1,
                "dayString": "29-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "30",
                "dayLink": "2018/10/30",
                "dayName": "Tuesday",
                "dayOfWeek": 2,
                "dayString": "30-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "day": "31",
                "dayLink": "2018/10/31",
                "dayName": "Wednesday",
                "dayOfWeek": 3,
                "dayString": "31-October-2018",
                "eventsForDay": Array [],
                "ongoingEventsForDay": Array [],
              },
              Object {
                "empty": true,
              },
              Object {
                "empty": true,
              },
              Object {
                "empty": true,
              },
              Object {
                "empty": true,
              },
            ],
          ]
        }
        clickDay={[Function]}
        openinghoursexceptions={Object {}}
        settings={
          Object {
            "activeLogo": "http://foo.com/image.png",
            "dateFormat": "DD.MM.YYYY",
            "pageSubtitle": "",
            "pageTitle": "",
            "subtitleTextColor": "#000",
            "theme": "darkly",
            "titleBarAngle": "150",
            "titleBarColor": "#000",
            "titleBarColor2": "#ec1c22",
            "titleBarPercentage": "50",
            "titleTextColor": "",
          }
        }
      />
    </div>
  </div>
</section>
`;
